testflinger_agent/job.py

```diff
@@ -133,85 +147,66 @@ def run_with_log(self, cmd, logfile, cwd=None):
         )
         global_timeout = self.get_global_timeout()
         output_timeout = self.get_output_timeout()
-        start_time = time.time()
-        with open(logfile, "a", encoding="utf-8") as f:
-            live_output_buffer = ""
-            readpoll = select.poll()
-            buffer_timeout = time.time()
-            process = subprocess.Popen(
-                cmd,
-                stdout=subprocess.PIPE,
-                stderr=subprocess.STDOUT,
-                shell=True,
-                cwd=cwd,
-                env=env,
-            )
-
-            def cleanup(signum, frame):
+        start_time = last_output_time = time.time()
+        process = subprocess.Popen(  # pylint: disable=consider-using-with
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            shell=True,
+            cwd=cwd,
+            env=env,
+        )
+        # Make sure we don't block on the stdout pipe
+        os.set_blocking(process.stdout.fileno(), False)
+
+        # Setup select.poll to check for output
+        readpoll = select.poll()
+        readpoll.register(process.stdout, select.POLLIN)
+
+        def cleanup(_, __):
```

Comment:
why don't just use `def cleanup(*args)`? Or even without params

Reply:
signal.signal() calls the handler function with two args, so it needs to accept those. But in our case, we don't actually care about doing anything with those args - so pylint complains about not using them. So *args won't work, but it gave me the idea to use *_ instead, which also makes pylint happy. So thanks!

Reply:
See, thanks!